daemon/emer-aggregate-tally.c

```diff
@@ -92,97 +103,131 @@ aggregate_timer_to_variant (guint32    unix_user_id,
-                                      bytes,
-                                      FALSE);
+#define CHECK(x) \
+  check_sqlite_error (self, (G_STRLOC), (x))
 
-  if (!g_variant_is_normal_form (variant))
-    {
-      g_autoptr(GError) deletion_error = NULL;
-      g_autoptr(GFile) file = NULL;
+static GVariant *
+column_to_variant (sqlite3_stmt *stmt,
+                   int           i)
+{
+  const void *text = sqlite3_column_text (stmt, i);
+  return g_variant_parse (NULL, text, NULL, NULL, NULL);
+}
 
-      g_warning ("Corrupted aggregate timer at %s, deleting...", path);
+static guint32
+column_to_uint32 (sqlite3_stmt *stmt,
+                  int           i)
+{
+  int number = sqlite3_column_int (stmt, i);
+  return *((guint32*) &number);
+}
 
-      file = g_file_new_for_path (path);
-      g_file_delete (file, NULL, &deletion_error);
-      if (deletion_error)
-        g_warning ("Error deleting corrupted tally file %s: %s",
-                   path, deletion_error->message);
-      return NULL;
+#if 0
+static void
+dump_tally (EmerAggregateTally *self)
+{
+  const char *SELECT_SQL = "SELECT * FROM tally";
+  sqlite3_stmt *stmt = NULL;
+  int ret;
+
+  CHECK (sqlite3_prepare_v2 (self->db, SELECT_SQL, -1, &stmt, NULL));
+
+  while ((ret = sqlite3_step (stmt)) == SQLITE_ROW)
+    {
+      g_autofree char* s_date = g_strdup ((const char*)sqlite3_column_text (stmt, 0));
+      g_autofree char* s_event_id = g_strdup ((const char*)sqlite3_column_text (stmt, 1));
+      guint32 s_unix_user_id = column_to_uint32 (stmt, 2);
+      g_autofree char* s_aggregate_key = g_strdup ((const char*)sqlite3_column_text (stmt, 3));
+      g_autofree char* s_payload = g_strdup ((const char*)sqlite3_column_text (stmt, 4));
+      guint32 s_counter = column_to_uint32 (stmt, 5);
+
+      g_print ("%s|%s|%u|%s|%s|%u\n",
+               s_date, s_event_id, s_unix_user_id,
+               s_aggregate_key, s_payload, s_counter);
     }
 
-  return g_steal_pointer (&variant);
+  CHECK (sqlite3_finalize (stmt));
 }
+#endif
 
-static guint32
-find_previous_aggregate_timer_counter (EmerAggregateTally *self,
-                                       const char         *aggregate_counter_path)
+static void
+emer_aggregate_tally_constructed (GObject *object)
 {
-  g_autoptr(GVariant) variant = NULL;
-  guint32 previous_counter = 0;
+  EmerAggregateTally *self = EMER_AGGREGATE_TALLY (object);
+  g_autofree gchar *path = NULL;
+  int ret;
 
-  variant = load_aggregate_timer_from_path (aggregate_counter_path);
+  G_OBJECT_CLASS (emer_aggregate_tally_parent_class)->constructed (object);
 
-  if (variant)
-    g_variant_get_child (variant, 3, "u", &previous_counter);
+  g_assert (self->persistent_cache_directory != NULL);
+  ensure_folder_exists (self, self->persistent_cache_directory, /* TODO */ NULL);
+  path = g_build_filename (self->persistent_cache_directory,
+                           "metrics.db",
```

Comment:
I was going to suggest that this be named `aggregate-events.db` or something since that's all it's used for, but I could see this gaining wider usage later.